server/src/services/TodoService.ts

```diff
@@ -4,25 +4,42 @@ import pool from "../../todosDb";
 interface GetTodosOptions {
   itemsPerPage?: number;
   page?: number;
+  name?: string;
+  types?: string[];
 }
 
 export const getTodosService = async ({
   itemsPerPage = 10,
   page = 1,
+  name = "",
+  types = [],
 }: GetTodosOptions = {}) => {
   const offset = (page - 1) * itemsPerPage;
-  const query = `SELECT * FROM todos LIMIT $1 OFFSET $2`;
-  const countQuery = `SELECT COUNT(*) FROM todos`;
+  const hasTypes = types.length > 0;
 
-  const result = await pool.query(query, [itemsPerPage, offset]);
-  const countResult = await pool.query(countQuery);
+  const query = `
+    SELECT * FROM todos
+    WHERE name ILIKE $3
+    ${hasTypes ? `AND type = ANY($4)` : ""}
```

Comment:
In the future, assigning numbers by the params array would be good. So, you insert in the array your param and then add a string with ${paramArraay.length} number. don't fix it now, but keep in mind